Lib/functools.py

```diff
@@ -273,43 +273,123 @@ def reduce(function, sequence, initial=_initial_missing):
 ### partial() argument application
 ################################################################################
 
+
+class __PlaceholderTypeBase:
+    """The type of the Placeholder singleton.
+
+    Used as a placeholder for partial arguments.
+    """
+    __instance = None
```

Comment:
```suggestion
    __instance = None
    __slots__ = ()
```

Add `__slots__` to avoid member assignment (to align with C implementation).